src/controls/Control.ts

```diff
@@ -169,11 +170,17 @@ export class Control {
    */
   declare mouseUpHandler?: ControlActionHandler;
 
-  shouldActivate(controlKey: string, fabricObject: InteractiveFabricObject) {
+  shouldActivate(
+    controlKey: string,
+    fabricObject: InteractiveFabricObject,
+    pointer: Point,
+    { tl, tr, br, bl }: TCornerPoint
+  ) {
     // TODO: locking logic can be handled here instead of in the control handler logic
     return (
       fabricObject.canvas?.getActiveObject() === fabricObject &&
-      fabricObject.isControlVisible(controlKey)
+      fabricObject.isControlVisible(controlKey) &&
+      Intersection.isPointInPolygon(pointer, [tl, tr, br, bl])
```

Comment:
I am very pleased from this.
Now if I want to create a control for scaling a side I can simply replace the last call with `intersectLinePolygon`